File: penguin_generator/tests/test_plugin/ios/Classes/TestPlugin.m
#import "TestPlugin.h"
#import "ChannelHandler+Generated.h"

@interface TestPlugin ()
@property ChannelHandler *handler;
@end

@implementation TestPlugin
- (instancetype)initWithChannel:(FlutterMethodChannel *)channel {
  self = [super init];
  if (self) {
    _handler = [[ChannelHandler alloc] initWithCallbackChannel:channel];
  }
  return self;
}

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"test_plugin"
            binaryMessenger:[registrar messenger]];
  TestPlugin* instance = [[TestPlugin alloc] initWithChannel:channel];
  [registrar addMethodCallDelegate:instance channel:channel];
  [registrar registerViewFactory:instance.handler.viewFactory withId:@"test_plugin/view"];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  [_handler.methodCallHandler onMethodCall:call result:result];
}

@end

File: packages/penguin_generator/tests/test_plugin/android/src/main/java/com/example/test_plugin/TestPlugin.java
package com.example.test_plugin;

import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** TestPlugin */
public class TestPlugin {
  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "test_plugin");
    final ChannelGenerated channelGenerated = new ChannelGenerated(channel);
    channel.setMethodCallHandler(channelGenerated.methodCallHandler);

    registrar
        .platformViewRegistry()
        .registerViewFactory("test_plugin/view", channelGenerated.viewFactory);
  }
}